pv_site_production/data/pv_data_sources.py

```diff
@@ -97,14 +96,22 @@ def get(
 
         _log.debug(f"Getting data from {start_ts} to {end_ts} for {len(site_uuids)} PVs")
         with self._database_connection.get_session() as session:  # type: ignore
-            generations = get_pv_generation_by_sites(
-                session=session,
-                start_utc=start_ts,
-                end_utc=end_ts,
-                # Convert to proper `UUID`s when we interact with the database.
-                site_uuids=[UUID(x) for x in site_uuids],
+            query = session.query(GenerationSQL).filter(
```

Comment:
Would it better to use a read function here? Both this component and the API will use this read function. Then we have less duplication and all the testing can be done in the datamodel repo

Reply:
I am trying to avoid common read/write functions. Each service that needs to have a direct access to the database will need something slightly different from it. It's a bad idea to enforce that they read/write it in the same way because whenever one needs something a little different, we need to change the common function.

Using a read function from another repo also obfuscate the code without much gain in readability, and it makes it way harder to debug.

It's also impossible to know exactly what a function like `read_generation_from_site_uuid(session, site_uuid, start_utc, end_utc)` does without looking at the code. Does it filter on both `start_utc`/`end_utc`, are bounds included or not, etc? And if those different slightly between the two places where we need to read from, we need an always more generic function.

Using sqlalchemy's ORM directly we get code that is clean, self-explanatory, easy to debug, and that can be easily changed without having to modify an external repo.